pkg/controller/constrainttemplate/stats_reporter.go

```diff
@@ -0,0 +1,158 @@
+package constrainttemplate
+
+import (
+	"context"
+	"time"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/metrics"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	ctCount        = "constraint_templates"
+	ingestCount    = "constraint_template_ingestion_count"
+	ingestDuration = "constraint_template_ingestion_duration_seconds"
+
+	ctCountDesc = "Number of observed constraint templates"
+)
+
+var (
+	ctCountM        = stats.Int64(ctCount, ctCountDesc, stats.UnitDimensionless)
+	ingestDurationM = stats.Float64(ingestDuration, "How long it took to ingest a constraint template in seconds", stats.UnitSeconds)
+
+	statusKey = tag.MustNewKey("status")
+
+	views = []*view.View{
+		{
+			Name:        ctCount,
+			Measure:     ctCountM,
+			Description: ctCountDesc,
+			Aggregation: view.LastValue(),
+			TagKeys:     []tag.Key{statusKey},
+		},
+		{
+			Name:        ingestCount,
+			Measure:     ingestDurationM,
+			Description: "Total number of constraint template ingestion actions",
+			Aggregation: view.Count(),
+			TagKeys:     []tag.Key{statusKey},
+		},
+		{
+			Name:        ingestDuration,
+			Measure:     ingestDurationM,
+			Description: "Distribution of how long it took to ingest a constraint template in seconds",
+			Aggregation: view.Distribution(0.01, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09, 0.1, 0.2, 0.3, 0.4, 0.5, 1, 2, 3, 4, 5),
+			TagKeys:     []tag.Key{statusKey},
+		},
+	}
+)
+
+type ctStatus string
+
+const (
+	statusActive = ctStatus("active")
```

Comment:
we already have `activeStatus` and `errorStatus`. can we reuse same naming conventions (either is fine) or share these (probably better)?

Reply:
Moved to a shared set of constants.